pkg/controllers/image/image.go

```diff
@@ -0,0 +1,330 @@
+package image
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/rancher/fleet/pkg/update"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Masterminds/semver/v3"
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	fleetcontrollers "github.com/rancher/fleet/pkg/generated/controllers/fleet.cattle.io/v1alpha1"
+	corev1controler "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/rancher/wrangler/pkg/kstatus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	lock sync.Mutex
+)
+
+const (
+	// AlphabeticalOrderAsc ascending order
+	AlphabeticalOrderAsc = "ASC"
+	// AlphabeticalOrderDesc descending order
+	AlphabeticalOrderDesc = "DESC"
+
+	defaultMessageTemplate = `Update from image update automation`
+)
+
+func Register(ctx context.Context, core corev1controler.Interface, gitRepos fleetcontrollers.GitRepoController, images fleetcontrollers.ImageScanController) error {
+	h := handler{
+		secretCache:  core.Secret().Cache(),
+		gitrepoCache: gitRepos.Cache(),
+	}
+
+	fleetcontrollers.RegisterImageScanStatusHandler(ctx, images, "ImageScanned", "image-scan", h.onChange)
+	return nil
+}
+
+type handler struct {
+	secretCache  corev1controler.SecretCache
+	gitrepoCache fleetcontrollers.GitRepoCache
+}
+
+func (h handler) onChange(image *v1alpha1.ImageScan, status v1alpha1.ImageScanStatus) (v1alpha1.ImageScanStatus, error) {
+	if image == nil || image.DeletionTimestamp != nil {
+		return status, nil
+	}
+
+	if image.Spec.Suspend {
+		return status, nil
+	}
```

Comment:
nit: You may be able to combine this conditionals.

Reply:
This is separated for code readibility.